### PR TITLE
Modifying to add an additional prop to clear the input if a selection…

### PR DESCRIPTION
**Pull request by jimbob687**
… is made, useful if wanting to do something with the search result other than add it to the input, e.g. add a new child element from a different component.
Prop is called "clearOnSelect"
Haven't tested this myself yet as hacked up the code in my own env and added as a component. Works in my own code.
Hope this is something that you will find useful

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -225,11 +228,17 @@ export default class ReactSearchBox extends Component {
...
+    if(this.props.clearOnSelect) {
+      // clear the value rather than input it into the box
+      value = "";
+    }
+
     const { onSelect, onChange } = this.props
 
     this.setState({
-      value,
+      value: value,
```
Comment (ghoshnirmalya) [inline src/index.js:241]: This change doesn't look necessary.

```suggestion
      value,
```

File: src/index.js [hunk 2]
```diff
@@ -225,11 +228,17 @@ export default class ReactSearchBox extends Component {
      * The 'value' state is updated with the clicked record's value.
      */
 
-    const { value } = record
+    let {value} = record;
+
+    if(this.props.clearOnSelect) {
+      // clear the value rather than input it into the box
+      value = "";
```
Comment (ghoshnirmalya) [inline src/index.js:235]: You need to modify the state like this:

```suggestion
      this.setState({
        value: ""      
      })
```

**Review by ghoshnirmalya — COMMENTED** [review]
@jimbob687 Thank you. I think that this would be useful. I can approve your PR once you address my review comments.

File: src/index.js (hunk 2 above)
Comment (jimbob687) [reply src/index.js:235]: Apologize as not the greatest at react, but set it to an empty string and the state will be updated below that. If I set the state there value will get set again in the next state call which will cause a race condition. Is that the correct thinking and if so, is there a smarter way to do this than what I'm doing now?

**Review by ghoshnirmalya — APPROVED** [review]

**Pull request merged**